Framework/Geometry/src/Instrument/DetectorInfo.cpp

```diff
@@ -321,34 +317,23 @@ const std::vector<detid_t> &DetectorInfo::detectorIDs() const {
 }
 
 /// Returns the scan count of the detector with given detector index.
-size_t DetectorInfo::scanCount(const size_t index) const {
-  return m_detectorInfo->scanCount(index);
-}
+size_t DetectorInfo::scanCount() const { return m_detectorInfo->scanCount(); }
 
 /** Returns the scan interval of the detector with given index.
  *
  * The interval start and end values would typically correspond to nanoseconds
  * since 1990, as in Types::Core::DateAndTime. */
-std::pair<Types::Core::DateAndTime, Types::Core::DateAndTime>
-DetectorInfo::scanInterval(const std::pair<size_t, size_t> &index) const {
-  const auto &interval = m_detectorInfo->scanInterval(index);
-  return {interval.first, interval.second};
-}
-
-/** Set the scan interval of the detector with given detector index.
- *
- * The interval start and end values would typically correspond to nanoseconds
- * since 1990, as in Types::Core::DateAndTime. Note that it is currently not
- *possible
- * to modify scan intervals for a DetectorInfo with time-dependent detectors,
- * i.e., time intervals must be set with this method before merging individual
- * scans. */
-void DetectorInfo::setScanInterval(
-    const size_t index,
-    const std::pair<Types::Core::DateAndTime, Types::Core::DateAndTime>
-        &interval) {
-  m_detectorInfo->setScanInterval(index, {interval.first.totalNanoseconds(),
-                                          interval.second.totalNanoseconds()});
+const std::vector<std::pair<Types::Core::DateAndTime, Types::Core::DateAndTime>>
+    &DetectorInfo::scanIntervals() const {
+  std::vector<std::pair<Types::Core::DateAndTime, Types::Core::DateAndTime>>
+      *intervals_vector = new std::vector<
+          std::pair<Types::Core::DateAndTime, Types::Core::DateAndTime>>(
+          (m_detectorInfo->scanIntervals()).size());
+  for (auto it = m_detectorInfo->scanIntervals().begin();
+       it != m_detectorInfo->scanIntervals().end(); ++it) {
```

Comment:
Can use range-based `for`:
```cpp
for ( const auto &interval : m_detectorInfo->scanIntervals())
  ...
```
But it looks like implicit conversion to `DateAndTime` is working, so try this instead:
```cpp
const auto &intervals = m_detectorInfo->scanIntervals();
return {intervals.begin(), intervals.end()};
```